roles/cockpit/templates/cockpit.yml.j2

```diff
@@ -0,0 +1,25 @@
+http:
+  routers:
+    {{ cockpit_name }}-http:
+      entryPoints:
+        - "web"
+      rule: "Host(`{{ cockpit_web_subdomain + '.' + cockpit_web_domain }}`)"
```

Comment:
Maybe try using traefik_host here as it should do the empty subdomain check for you.